hemework25-datestr.py

```diff
@@ -0,0 +1,160 @@
+#TASK1
+from functools import wraps
+def check_full(method):
+    @wraps(method)
+    def wrapper(self, *args, **kwargs):
+        if self.is_full():
+            raise ValueError("Storage is full.")
+        return method(self, *args, **kwargs)
+    return wrapper
+
+def check_empty(method):
+    @wraps(method)
+    def wrapper(self, *args, **kwargs):
+        if self.is_empty():
+            raise ValueError("Storage is empty.")
+        return method(self, *args, **kwargs)
+    return wrapper
+
+class BaseSequence:
+    """A base sequence class."""
+    def __init__(self, initial_items: list, capacity=10):
+        self._storage = initial_items
+        self._capacity = capacity
+
+    @check_empty
+    def put(self, element):
+        """Put an element into the sequence."""
+        if self.is_full():
+            raise ValueError("Storage is full.")
+        self._storage.append(element)
+
+    def is_empty(self):
+        """Check if the sequence is empty."""
+        return not bool(self._storage)
+
+    def is_full(self):
+        """Check if the sequence is full."""
+        return len(self._storage) == self._capacity
+
+
+class Stack(BaseSequence):
+    """A Last-In-First-Out (LIFO) data structure."""
+    @check_empty
+    def pop(self):
+        """Remove and return the last element."""
+        if self.is_empty():
```

Comment:
Again, you already decorated the method with the `check_empty` decorator which does this. You don't need this check anymore.